ports/esp32/machine_pwm.c

```diff
@@ -316,23 +329,27 @@ STATIC void set_duty_u16(machine_pwm_obj_t *self, int duty) {
     if ((duty < 0) || (duty > UI_MAX_DUTY)) {
         mp_raise_msg_varg(&mp_type_ValueError, MP_ERROR_TEXT("duty_u16 must be from 0 to %d"), UI_MAX_DUTY);
     }
-    duty >>= HIGHEST_PWM_RES + UI_RES_SHIFT - timers[TIMER_IDX(self->mode, self->timer)].duty_resolution;
-    int max_duty = (1 << timers[TIMER_IDX(self->mode, self->timer)].duty_resolution) - 1;
-    if (duty < 0) {
-        duty = 0;
-    } else if (duty > max_duty) {
-        duty = max_duty;
-    }
-    check_esp_err(ledc_set_duty(self->mode, self->channel, duty));
+    ledc_timer_config_t timer = timers[TIMER_IDX(self->mode, self->timer)];
+    int _duty = duty >> (HIGHEST_PWM_RES + UI_RES_SHIFT - timer.duty_resolution);
```

Comment:
Identifiers starting with `_` are reserved for the system, please don't use them.